packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -0,0 +1,280 @@
+import {
+  TransformerResolverProvider,
+  DataSourceProvider,
+  TransformerContextProvider,
+  TransformerResolversManagerProvider,
+  AppSyncFunctionConfigurationProvider,
+  MappingTemplateProvider,
+  GraphQLApiProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { Stack, isResolvableObject } from '@aws-cdk/core';
+
+import { MappingTemplate, S3MappingTemplate } from '../cdk-compat';
+import { StackManager } from './stack-manager';
+import assert from 'assert';
+import { toPascalCase } from 'graphql-transformer-common';
+import { dedent } from 'ts-dedent';
+
+type Slot = {
+  requestMappingTemplate: MappingTemplateProvider;
+  responseMappingTemplate?: MappingTemplateProvider;
+  dataSource?: DataSourceProvider;
+};
+
+// Name of the None Data source used for pipeline resolver
+const NONE_DATA_SOURCE_NAME = 'NONE_DS';
+
+export class ResolverManager implements TransformerResolversManagerProvider {
```

Comment:
Provider suffix seems extra. Since it is implementing a resolver manager, not a provider that will product a resolver manager instance. In general looking at the other Providers perhaps it would be useful just to prefix these interfaces with `I`.